### PR TITLE
Added support for big tensor pooling

### PR DESCRIPTION
**Pull request by kaixih**
Cudnn legacy API only supports int32 indexing and can handle a maximum of 2^31-1 elements. This PR added support of pooling operations over big tensor. In particular, we split the big tensor along the batch axis when necessary and call cudnn API sequentially.

cc. @nluehr

### CODE REVIEW

**Comment by gbaned** [comment]
@jurahul Can you please review this PR ? Thank you!

**Comment by nluehr** [comment]
Gentle ping @jurahul

**Comment by gbaned** [comment]
@kaixih Can you please resolve conflicts? Thank you!

**Comment by kaixih** [comment]
Conflict resolved. PTAL.

File: tensorflow/stream_executor/cuda/cuda_dnn.cc [hunk 1]
```diff
@@ -6009,6 +6009,64 @@ bool CudnnSupport::DoActivate(Stream* stream,
...
 }
 
+namespace {
+// Cudnn legacy API only supports int32 indexing and can handle a maximum of
+// 2^31-1 elements. For pooling operations, we split the big tensor along the
+// batch axis if necessary and call cudnn API sequentially.
+// This function will return a vector of tuples, representing how to split the
+// tensors: (batch number of each split, offset in bytes of the input, offset in
+// bytes of the output).
+port::StatusOr<std::vector<std::tuple<int64_t, int64_t, int64_t>>>
```
Comment (pjannaty) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc:6019]: Prefer structs over tuples. See [style guide](https://google.github.io/styleguide/cppguide.html#Structs_vs._Tuples)

Comment (kaixih) [reply tensorflow/stream_executor/cuda/cuda_dnn.cc]: Thanks. I have pushed new commit to resolve this. PTAL.

Comment (pjannaty) [reply tensorflow/stream_executor/cuda/cuda_dnn.cc]: Thanks, Kaixi! Reads much better!

**Review by gcforster — APPROVED** [review]

**Pull request merged**